src/Nest/CommonOptions/TimeUnit/Time.cs

```diff
@@ -20,7 +21,7 @@ public class Time : IComparable<Time>, IEquatable<Time>
 		private const double MicrosecondsInAMillisecond = 10;
 
 		private static readonly Regex ExpressionRegex =
-			new Regex(@"^(?<factor>[-+]?\d+(?:\.\d+)?)\s*(?<interval>(?:y|w|d|h|m|s|ms|nanos|micros))?$",
+			new Regex(@"^(?<factor>[-+]?[\d\.e\-]+?)\s*(?<interval>(?:y|w|d|h|m|s|ms|nanos|micros))?$",
```

Comment:
Cool i thought those were invalid case 👍 

Only have one small addition to the regex:

`(?!^)\s{0,10}` to prevent `<leading spaces>nanos` being a match.